src/sentry/models/exporteddata.py

```diff
@@ -57,18 +57,20 @@ def status(self):
         else:
             return ExportStatus.Valid
 
-    @property
-    def date_expired_string(self):
-        if self.date_expired is None:
-            return None
-        return self.date_expired.strftime("%-I:%M %p on %B %d, %Y (%Z)")
-
     @property
     def payload(self):
         payload = self.query_info.copy()
         payload["export_type"] = ExportQueryType.as_str(self.query_type)
         return payload
 
+    def get_date_string(self, date_field):
+        if not hasattr(self, date_field):
+            raise ValueError(u"Unrecognized field {} on ExportedData object".format(date_field))
```

Comment:
@leeandher It looks like `finalize_upload` is going to blow up if you raise this value error. Does it make sense to have a try/except for this case?

Reply:
I refactored it a bit to avoid throwing the error. It's an internal helper so it just passes on an AttributeError, even if it'll never happen.